core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
FWIW I did it here: https://github.com/kayousterhout/spark-1/commit/fdab2998fd52b3bb9592d30bafb96f9de98af6c9 a while ago in my testing of the other PR (although I wouldn't do it quite this way -- I'd pass in the serialized task as an argument to TaskDescription's encode() method, to avoid having to change the LaunchTask def., which requires Mesos changes)